File: bestlocationstrategy/src/main/java/com/location/bestlocationstrategy/LocationChangesListener.java
package com.location.bestlocationstrategy;

import android.location.Location;

/**
 * Created by Balwinder on 20/Sept/2018.
 */

public interface LocationChangesListener {
    void onLocationChanged(Location location);
    void onConnected();
    void onConnectionStatusChanged();
    void onFailure(String failureMessage);
}

File: bestlocationstrategy/src/main/java/com/location/bestlocationstrategy/LocationChangesListener.kt
package com.location.bestlocationstrategy

import android.location.Location
import com.location.bestlocationstrategy.utils.Error

/**
 * Created by Balwinder on 20/Sept/2018.
 */
interface LocationChangesListener {
    fun onBetterLocationAvailable(location: Location?)
    fun onConnected()
    fun onConnectionStatusChanged()
    fun onFailure(error: Error)
}